### PR TITLE
GPII-1754: Update to newest express (including new configuration options and priority info)...

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [GPII-1754](https://issues.gpii.net/browse/GPII-1754) for details.

### CODE REVIEW

File: src/js/server/api.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
 /*
 
-  Mount all the feature-specific endpoints under a single umbrella class.  Returns the API documentation by default.
+    Mount all the feature-specific endpoints under a single umbrella class.  Returns the API documentation by default.
+
+    You must have all of the following middleware loaded before the routers in this module:
```
Comment (amb26) [inline src/js/server/api.js]: What do you mean by "having all of the following middleware loaded"?

File: src/js/server/api.js [hunk 2]
```diff
@@ -142,27 +116,54 @@ fluid.defaults("gpii.express.user.api", {
...
         }
     }
 });
 
-// An instance of `gpii.express.user.api` that already has the required session middleware wired in
-//
-// Generally you will want to start with the base grade and provide your own.
-fluid.defaults("gpii.express.user.api.hasMiddleware", {
-    gradeNames: ["gpii.express.user.api"],
+// A mix-in grade to add the required session middleware.  You can mix this in with the api itself, but generally you
```
Comment (amb26) [inline src/js/server/api.js]: I find this comment quite unclear. What does "the required session middleware" do? What does the "loginRequired" grade do?

**Comment by the-t-in-rtf** [comment]
Now that we are trending towards approving the `gpii-express` pull, and now that I have got `gpii-json-schema` working again with the latest updates, I can finally work on this.  However, it is not safe for further review at the moment. I will comment here on the outstanding issues and ping you when it is ready.

**Comment by the-t-in-rtf** [comment]
@amb26, I have updated this package to the latest `gpii.express` and pushed through various best practices discussed in other PRs.  This should be safe to begin reviewing.

File: src/js/server/api.js (hunk 1 above)
Comment (the-t-in-rtf) [inline src/js/server/api.js]: I have added documentation for this in `apiComponent.md`.  Please review.

**Comment by the-t-in-rtf** [comment]
This is ready for review, but should not be merged/published until its (manifold) dependencies are merged/published:

* [gpii-binder](https://github.com/GPII/gpii-binder/pull/3)
* [gpii-express](https://github.com/GPII/gpii-express/pull/25)
* [gpii-pouchdb](https://github.com/GPII/gpii-pouchdb/pull/12)
* gpii-mail-test needs to be published.
* [gpii-webdriver](https://github.com/GPII/gpii-webdriver/pull/1)
* [gpii-handlebars](https://github.com/GPII/gpii-handlebars/pull/20)
* [gpii-json-schema](https://github.com/GPII/gpii-json-schema/pull/5)

**Comment by the-t-in-rtf** [comment]
My comment was optimistic, as I was still working on converting to webdriver until just now.  It is genuinely ready now, although it should be very far down on the list of priorities.

**Comment by the-t-in-rtf** [comment]
When we get around to this, we should review all-tests.js together.  All of the tests required from all-tests.js pass individually, but fail when run together.  I suspect a similar problem in gpii-json-schema as well.

**Pull request merged**